src/plugins/console/public/application/containers/editor/monaco/monaco_editor_actions_provider.ts

```diff
@@ -347,6 +403,50 @@ export class MonacoEditorActionsProvider {
     return this.getSuggestions(model, position, context);
   }
 
+  /*
+   * This function inserts a request from the history into the editor
+   */
+  public async restoreRequestFromHistory(request: string) {
+    const model = this.editor.getModel();
+    if (!model) {
+      return;
+    }
+    let position = this.editor.getPosition() as monaco.IPosition;
+    const requests = await this.getSelectedParsedRequests();
+    let prefix = '';
+    let suffix = '';
+    // if there are requests at the cursor/selection, insert either before or after
+    if (requests.length > 0) {
+      // if on the 1st line of the 1st request, insert at the beginning of that line
+      if (position && position.lineNumber === requests[0].startLineNumber) {
+        position = { column: 1, lineNumber: position.lineNumber };
+        suffix = '\n';
+      } else {
+        // otherwise insert at the end of the last line of the last request
+        const lastLineNumber = requests[requests.length - 1].endLineNumber;
+        position = { column: model.getLineMaxColumn(lastLineNumber), lineNumber: lastLineNumber };
+        prefix = '\n';
+      }
+    } else {
+      // if not inside a request, insert the request at the cursor position
```

Comment:
As far as I understand, the request from history will always end up at column 1 unless it is not inside a request - then it might end up in the middle of the line if the cursor is there. In Ace editor, it always ends up at column 1. Should we do the same here for consistency?

```suggestion
      // the cursor is not inside a request
      if (!position) {
        // if no cursor position, insert at the beginning of the first line
        position = { lineNumber: 1, column: 1 };
      } else {
        // insert the request at the beginning of the line of the cursor position
        position = { lineNumber: position.lineNumber, column: 1 };
      }
```

In Ace editor:

https://github.com/elastic/kibana/assets/59341489/39eff9b0-e956-4796-b598-f7713dd6b98b




In this PR:

https://github.com/elastic/kibana/assets/59341489/89d5c016-018d-4609-b51f-084cc105fb65

Reply:
Thanks a lot, @ElenaStoeva! Another great catch, fixed in https://github.com/elastic/kibana/pull/183181/commits/67968c6b89b6c70d764f7a9c0850fd8c20e6b63c